### PR TITLE
Fix issues with partial shared schema init

### LINKED ISSUE #2534: requiredProperties unrecognized selector sent to class when using two realms with one using Class subsets and one is bundled read only realm
Hi,

I have two realms and two classes: t1.realm is the read-write main realm and only persists class t1, t2.realm is the bundled read only realm generated with only class t2. The class definition of t1 and t2 are as follows:

class t1: Object {
  dynamic var f1: String = ""
  dynamic var f2: String = ""
}

class t2: Object {
  dynamic var f1: String = ""
  dynamic var f2: String = ""
}

I can successfully open t1.realm with the following code:

let t1Config = Realm.Configuration(path: NSTemporaryDirectory().stringByAppendingString("t1.realm"), objectTypes: [t1.self])
let t1Realm = Realm(configuration: t1Config)!

However, when I open t2.realm with the following code:

// For testing purpose, I do not put t2.realm in the app bundle
let t2Config = Realm.Configuration(path: NSTemporaryDirectory().stringByAppendingString("t2.realm"), readOnly: true)
let t2Realm = Realm(configuration: t2Config)!

App stopped with the following stack trace:

2015-09-16 10:44:27.636 RealmTest[29697:678644] +[RLMAccessor_v0_t1 requiredProperties]: unrecognized selector sent to class 0x7faf0bc27530
2015-09-16 10:44:27.644 RealmTest[29697:678644] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[RLMAccessor_v0_t1 requiredProperties]: unrecognized selector sent to class 0x7faf0bc27530'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001010fac65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x0000000102c65bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x0000000101101fad +[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x000000010105813c ___forwarding___ + 988
	4   CoreFoundation                      0x0000000101057cd8 _CF_forwarding_prep_0 + 120
	5   Realm                               0x0000000100c7a709 +[RLMObjectSchema propertiesForClass:isSwift:] + 1850
	6   Realm                               0x0000000100c793f8 +[RLMObjectSchema schemaForObjectClass:] + 421
	7   Realm                               0x0000000100cb84a1 +[RLMSchema registerClasses:count:] + 729
	8   Realm                               0x0000000100cb93ac _ZNSt3__117__call_once_proxyINS_5tupleIJZ25+[RLMSchema sharedSchema]E3$_0EEEEEvPv + 93
	9   libc++.1.dylib                      0x00000001032a4a42 _ZNSt3__111__call_onceERVmPvPFvS2_E + 117
	10  Realm                               0x0000000100cb883e +[RLMSchema sharedSchema] + 53
	11  Realm                               0x0000000100caad03 +[RLMRealm realmWithConfiguration:error:] + 1115
	12  RealmSwift                          0x0000000100f4a272 _TFC10RealmSwift5RealmCfMS0_FT13configurationVS0_13Configuration5errorGVSs33AutoreleasingUnsafeMutablePointerGSqCSo7NSError___GSqS0__ + 722
	13  RealmTest                           0x0000000100babd7d _TFC9RealmTest14ViewController11viewDidLoadfS0_FT_T_ + 3341
	14  RealmTest                           0x0000000100babfc2 _TToFC9RealmTest14ViewController11viewDidLoadfS0_FT_T_ + 34
	15  UIKit                               0x0000000101aca1d0 -[UIViewController loadViewIfRequired] + 738
	16  UIKit                               0x0000000101aca3ce -[UIViewController view] + 27
	17  UIKit                               0x00000001019e5289 -[UIWindow addRootViewControllerViewIfPossible] + 58
	18  UIKit                               0x00000001019e564f -[UIWindow _setHidden:forced:] + 247
	19  UIKit                               0x00000001019f1de1 -[UIWindow makeKeyAndVisible] + 42
	20  UIKit                               0x0000000101995417 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2732
	21  UIKit                               0x000000010199819e -[UIApplication _runWithMainScene:transitionContext:completion:] + 1349
	22  UIKit                               0x0000000101997095 -[UIApplication workspaceDidEndTransaction:] + 179
	23  FrontBoardServices                  0x000000010568b5e5 __31-[FBSSerialQueue performAsync:]_block_invoke_2 + 21
	24  CoreFoundation                      0x000000010102e41c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	25  CoreFoundation                      0x0000000101024165 __CFRunLoopDoBlocks + 341
	26  CoreFoundation                      0x0000000101023f25 __CFRunLoopRun + 2389
	27  CoreFoundation                      0x0000000101023366 CFRunLoopRunSpecific + 470
	28  UIKit                               0x0000000101996b02 -[UIApplication _run] + 413
	29  UIKit                               0x00000001019998c0 UIApplicationMain + 1282
	30  RealmTest                           0x0000000100bb2037 main + 135
	31  libdyld.dylib                       0x000000010414c145 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException

### PR DESCRIPTION
**Pull request by tgoyne**
When the initial partial schema init is done with an explicit class list, duplicate object schemas for those classes would end up in the final shared schema, along with object schemas for the accessor classes. No new tests for any of this because we can't test it via our testing framework....

Closes #2534.

@jpsim @bdash

### CODE REVIEW

**Comment by bdash** [comment]
:+1:

File: Realm/RLMAccessor.mm [hunk 1]
```diff
@@ -597,6 +597,21 @@ void RLMReplaceSharedSchemaMethod(Class accessorClass, RLMObjectSchema *schema)
     class_replaceMethod(metaClass, @selector(sharedSchema), imp, "@@:");
 }
 
+// Declare a selector that is only implemented by generated classes
+@interface RLMDummyDeclaration : NSObject
++ (void)RLMIsGeneratedClass;
+@end
+
+static void RLMMarkClassAsGenerated(Class cls) {
```
Comment (jpsim) [inline Realm/RLMAccessor.mm]: Why not store a vector of generated classes instead? Then you could check for its presence in the vector in `RLMIsGeneratedClass` without the need to use the Objective-C runtime or the dummy class & selector.

**Comment by tgoyne** [comment]
Added a test for initializing the shared schema after the partial schema using the multiprocess test stuff.

**Comment by jpsim** [comment]
Great! :+1: once this passes CI. Currently failing with:

```
In file included from /Users/realm/workspace/objc_pr/configuration/Release/swift_version/1.2/target/ios7-static/tightdb_objc/Realm/Tests/SchemaTests.mm:21:
/Users/realm/workspace/objc_pr/configuration/Release/swift_version/1.2/target/ios7-static/tightdb_objc/Realm/Tests/RLMMultiProcessTestCase.h:29:4: error: expected a type
- (NSTask *)childTask;
   ^
1 error generated.
```

**Pull request merged**